inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp

```diff
@@ -810,29 +819,44 @@ void MKLDNNGraph::Infer(MKLDNNInferRequest* request, int batch) {
 
     ENABLE_CPU_DEBUG_CAP(NodeDumper nd(config.debugCaps, infer_count));
 
-    for (int i = 0; i < graphNodes.size(); i++) {
-        if (request != nullptr) {
+#ifdef CPU_DEBUG_CAPS
+    for (const auto& node : constantGraphNodes) {
+        if (request != nullptr)
             request->ThrowIfCanceled();
-        }
 
-        PERF(graphNodes[i]);
+        ENABLE_CPU_DEBUG_CAP(nd.dumpInputBlobs(node));
+        ENABLE_CPU_DEBUG_CAP(nd.dumpOutputBlobs(node));
+    }
+#endif
+
+    for (const auto& node : mutableGraphNodes) {
+        if (request != nullptr)
+            request->ThrowIfCanceled();
 
         if (batch > 0)
-            graphNodes[i]->setDynamicBatchLim(batch);
+            node->setDynamicBatchLim(batch);
 
-        ENABLE_CPU_DEBUG_CAP(nd.dumpInputBlobs(graphNodes[i]));
+        ENABLE_CPU_DEBUG_CAP(nd.dumpInputBlobs(node));
 
-        if (!graphNodes[i]->isConstant()) {
-            OV_ITT_SCOPED_TASK(itt::domains::MKLDNNPlugin, graphNodes[i]->profiling.execute);
-            graphNodes[i]->execute(stream);
-        }
+        ExecuteNode(node, stream);
 
-        ENABLE_CPU_DEBUG_CAP(nd.dumpOutputBlobs(graphNodes[i]));
+        ENABLE_CPU_DEBUG_CAP(nd.dumpOutputBlobs(node));
     }
 
     if (infer_count != -1) infer_count++;
 }
 
+inline void MKLDNNGraph::ExecuteNode(const MKLDNNNodePtr& node, const mkldnn::stream& stream) const {
+    if (config.collectPerfCounters) {
+        PERF(node);
```

Comment:
@a-sidorova  @maxnick @EgorDuplensky still think we need to measure performance of the whole cycle content (including ThrowIfCanceled and etc). This overhead we pay for each node and it is CPU plug-in specific so logically should be included into pc. Also sum of perf counters has implicit requirement to be as much as close to resulting latency measured outside of infer call.

Reply:
Technically this overhead is a constant addition to the node execution time, which is, with no doubts, important to measure but does not help to understand the node execution time in itself. Moreover, it can even  blur node execution time if the overhead far outweighs  the node time. So practically we need latency measurement, which of course considers this overhead for the whole inference, but also we need precise execution time for each node in itself.

Reply:
Agree with @maxnick.
Also having node execution time as clean as possible even helps to measure the overhead, as Dmitry mentioned:
_Total Overhead = Latency - SumOfNodeExecutions._

Reply:
As it was stated many times: both metrics are important. The question is what we expose as default number. As soon as perf counters is part of public API and frequently used by customers to get brief performance information I would prefer to include overheads into calculation (actually this is honest numbers which OV takes for each layer execution). 
Task to distinguish between overheads and useful computations is more developer-specific. That's why is to put such counters under DEBUG_CAPS.

Reply:
From my perspective measurement of per layer execution time is already developer-specific and Latency is for the customers.
But maybe I am wrong.
Anyway, Dmitry is the one to decide after having all the arguments provided :)

Reply:
Than the decision is to measure whole loop content by default. Overhead is to be measured under DEBUG_CAPS (separate PR). Further discussions lets move offline.

Reply:
Well, I added measure whole loop content using `unique_ptr`. And I didn't notice perf regression due to the creation of `unique_ptr`